Assign Ability Scores Functionality, Closes #1

ashleawalker29 opened the pull request.

ashleawalker29 committed:
Add a .gitignore

ashleawalker29 committed:
Init the character class.

The character class to which all abilities will be assigned.

ashleawalker29 committed:
Init a dice roller.

ashleawalker29 committed:
Init constants file.

This includes printing constants, ability name constants, and an unchanging module for clearing the screen.

ashleawalker29 committed:
Step 1 -- Generation Type 1.

Implement support for generating scores and applying them in the order they were generated.

ashleawalker29 committed:
Step 1, Generation Type 1 -- UnitTest

ashleawalker29 committed:
Step 1, Generation Type 2 Functionality

Added functionality.
Added test coverage.

ashleawalker29 committed:
Step 1, Generation Type 3 Functionality.

Added generation type 3 functionality.
Added test coverage.

cnodland commented on steps/step1.py:
```diff
@@ -0,0 +1,141 @@
+
+        # Check input against possible abilities. Make sure we don't override any abilities that
+        # have just been set.
+        if (ability in (COM.lower(), 'com', '1') and
+                person.abilities[COM] is None):
+            apply_score(person, COM, scores)
+        elif (ability in (CON.lower(), 'con', '2') and
+              person.abilities[CON] is None):
+            apply_score(person, CON, scores)
+        elif (ability in (CUN.lower(), 'cun', '3') and
+              person.abilities[CUN] is None):
+            apply_score(person, CUN, scores)
+        elif (ability in (DEX.lower(), 'dex', '4') and
+              person.abilities[DEX] is None):
+            apply_score(person, DEX, scores)
+        elif (ability in (MAG.lower(), 'mag', '5') and
+              person.abilities[MAG] is None):
+            apply_score(person, MAG, scores)
+        elif (ability in (PER.lower(), 'per', '6') and
+              person.abilities[PER] is None):
+            apply_score(person, PER, scores)
+        elif (ability in (STR.lower(), 'str', '7') and
+              person.abilities[STR] is None):
+            apply_score(person, STR, scores)
+        elif (ability in (WIL.lower(), 'wil', 'will', '8') and
+              person.abilities[WIL] is None):
+            apply_score(person, WIL, scores)
+
```
I'm not the hugest fan of this giant else statement but off the top of my head i can't think of a more elegant way to write this, but i think it is worth taking a look at down the line

ashleawalker29 replied:
I've noted it. I totally agree.

cnodland commented on tests/step1_tests.py:
```diff
@@ -0,0 +1,57 @@
+
+        for ability_score in self.person.abilities.values():
+            self.assertTrue(ability_score is not None)
+            self.assertTrue(0 <= ability_score <= 3)
```
I like the tests but im not sure what "Step1" refers to in them

ashleawalker29 replied:
It does read kind of redundantly since it's under the class Step1Tests... I will change this.

cnodland reviewed (approved):
Nice start, im just not sure what to do with all of these big ol' elif statements

ashleawalker29 commented on steps/step1.py:
```diff
@@ -0,0 +1,141 @@
+            advancement_points = increment_score(person, CON, advancement_points)
+        elif (ability in (CUN.lower(), 'cun', '3') and person.abilities[CUN] < 3):
+            advancement_points = increment_score(person, CUN, advancement_points)
+        elif (ability in (DEX.lower(), 'dex', '4')and person.abilities[DEX] < 3):
```
```suggestion
        elif (ability in (DEX.lower(), 'dex', '4')and person.abilities[DEX] < 3):
        elif (ability in (MAG.lower(), 'mag', '5')and person.abilities[MAG] < 3):
        elif (ability in (PER.lower(), 'per', '6')and person.abilities[PER] < 3):
        elif (ability in (STR.lower(), 'str', '7')and person.abilities[STR] < 3):
        elif (ability in (STR.lower(), 'str', '7')and person.abilities[STR] < 3):
```

Spacing between ')and' needs to be ') and'.

ashleawalker29 committed:
Updated suggested changes.

Appeased pylint.
Corrected the name for an '__init__.py'

Added nested functions to improve readability.

ashleawalker29 committed:
Update step1_tests.py

cnodland reviewed (approved):
MUCH better 👍 way cleaner and easier to read!!!
